File: Data Structures & Algorithms/anagram-groups/submission-3.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        int n = strs.size();
        vector<string> temp(strs.begin(),strs.end());
        for(int i = 0;i<n;i++){
            sort(temp[i].begin(),temp[i].end());
        }
        unordered_map<string,vector<int>> mp;
        for(int i = 0;i<n;i++){
            mp[temp[i]].push_back(i);
        }

        vector<vector<string>> ans;
        for(auto &it:mp){
            vector<string> curr;
            for(int idx:it.second){
                curr.push_back(strs[idx]);
            }
            ans.push_back(curr);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/sum-of-two-integers/submission-3.java
class Solution {
    public int getSum(int a, int b) {
        // int carry = 0;
        int ans = 0;
        do{
            ans = (a^b);
            int carry = (a&b);
            carry = (carry<<1);
            a = ans;
            b = carry;
        }while(b != 0);
        return ans;
    }
}
